[RDY] init gpg python bindings +  Alot: 0.5.1 -> 0.7.0

teto opened the pull request:
###### Motivation for this change
Updating the CLI notmuch-based mailreader alot

###### Things done
- copied the extra/alot.desktop file to $out/share/applications
- now generates the man
- alot now needs the gpg binary for its tests, added it in checkOutputs.

Nevertheless some things are wrong:
- `man alot`doesn't work after installation via nix-env. I believe it's a known nixos problem. I've tried using multiple outputs but not sure how to make it better.
- 3 of the tests don't pass see https://github.com/NixOS/nixpkgs/issues/30392

I've not tested yet the gpg support but other than that it seems to work 

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Mic92 commented on pkgs/development/python-modules/gpg/default.nix:
```diff
@@ -1,18 +1,13 @@
-with pythonPackages;
-
-
-
```
This change uncovers that the library still fails to build on python3. But I have not investigated the reason for that.

teto commented:
2nd failure seems to be unrelated to PR (network failure). The one on macos I am not sure why it fails on why it installs this peculiar set of applications . How can I run that test locally ?
```
rm: missing operand

Try 'rm --help' for more information.

builder for ‘/nix/store/l0388f94ajl8shsjnnmj5jqyxdvx23cg-gpgme-1.9.0.drv’ failed with exit code 123

error: build of ‘/nix/store/l0388f94ajl8shsjnnmj5jqyxdvx23cg-gpgme-1.9.0.drv’ failed

The invocation of "nix-build -A nasty -A sylpheed -A python27Packages.notmuch -A libsForQt5.qgpgme -A python36Packages.afew -A notmuch -A gpgstats -A gmime3 -A sup -A gpgme -A grepm -A python36Packages.notmuch -A muchsync -A neomutt -A dar -A mutt -A python27Packages.afew -A mutt-with-sidebar -A minikube -A python36Packages.pygpgme -A python27Packages.pygpgme -A libsForQt56.qgpgme /Users/travis/.nox/nixpkgs" failed
```

Mic92 commented:
You would have to run the following command:

```
nix-build -I nixpkgs=path/to/nixpkgs -A nasty -A sylpheed -A python27Packages.notmuch -A libsForQt5.qgpgme -A python36Packages.afew -A notmuch -A gpgstats -A gmime3 -A sup -A gpgme -A grepm -A python36Packages.notmuch -A muchsync -A neomutt -A dar -A mutt -A python27Packages.afew -A mutt-with-sidebar -A minikube -A python36Packages.pygpgme -A python27Packages.pygpgme -A libsForQt56.qgpgme
```

However it fails on macOS. If you cannot test it there, I would suggest to restrict `meta.platforms` to linux for the moment for gpg python bindings.

teto commented on pkgs/development/python-modules/gpg/default.nix:
```diff
@@ -0,0 +1,40 @@
+    homepage = https://www.gnupg.org;
+    description = "GPG bindings for python";
+    license = licenses.lgpl2;
+    platforms = platforms.linux;
```
platform is already restricted to linux isn't it ?

Mic92 replied:
right, then `gpgme` is broken on darwin for unrelated reasons and we can ignore this build on travis.

Mic92 commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,60 @@
+      ];
+
+    doCheck = false;
+    postBuild= stdenv.lib.optionalString withManpage "make -C docs man";
```
rather then a build option, I would add `outputs = ["out" "man"]` to split the build and move the generated documentation there.

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,60 @@
+with pythonPackages;
+
+buildPythonApplication rec {
+    rev = "0.6";
```
indentation

FRidh replied:
rev -> version

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,60 @@
+{ stdenv, pkgs, pythonPackages
```
do not use `pkgs`

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,60 @@
+      ];
+
+    doCheck = false;
+    postBuild= stdenv.lib.optionalString withManpage "make -C docs man";
```
space before =

teto replied:
done

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,60 @@
+
+      sed "s,/usr/bin,$out/bin,g" extra/alot.desktop > $out/share/applications/alot.desktop
+      wrapProgram $out/bin/alot \
+        --prefix LD_LIBRARY_PATH : '${pkgs.lib.makeLibraryPath [ pkgs.notmuch pkgs.file pkgs.gpgme ]}'
```
Can you not patch the sources to hardcode the paths instead?

teto replied:
why ? this is a part I haven't touched so I would rather not change it as it works

FRidh commented on pkgs/development/libraries/gpgme/default.nix:
```diff
@@ -4,7 +4,12 @@
+  version = "1.9.0";
+
+  passthru = {
+    pythonSourceRoot = "lang/python";
```
is this worth it, considering you only use it for one derivation?

teto replied:
If gpgme next updates changes the path, it makes more sense to change the path in gpgme derviation then in the related ones imo. No strong opinion though.

teto replied:
While looking at clang package for an unrelated problem, I noticed it had a "python" output . Maybe I should do the same here instead ?
```
    outputs = [ "out" "python" ]
      ++ stdenv.lib.optional enableManpages "man";
```

FRidh replied:
Can the Python bindings be build without building gpgme? If so, then a separate derivation. If not, then an output is typically preferred to reduce building time. It is not required however. I'd say leave it as it is.

Anyway, please remove `pythonSourceRoot`.

teto replied:
I removed pythonSourceRoot and rebased

teto commented:
I've addressed your comments locally . Should I squash the history into to 2 commits ("gpg init" then "alot update") ?

teto commented:
I've tried to adress the review & squashed the commits. I tested (signature mode) the gpg support of alot and it worked fine.

teto commented:
Meson seems to keep failing but it's not linked to this PR

teto commented:
should I disable the mac platform ? I don't see where is the wrong rm call
```
rm: missing operand
Try 'rm --help' for more information.
rm: missing operand
Try 'rm --help' for more information.
```
and I can't test it on mac.

teto commented:
for my future self or anyone willing to take on; 
1/ there is actually no flag to disable gnupg doc so either add makeinfo dependency or patch Makefile.
2/ The generation of binding requires a swap in pip build order but possibly also in install order (not considered by upstream) thus such a patch might ne necessary (and upstreamed hopefully) https://github.com/teto/gpgme/commit/35c3d94d0b4395d36b37c20b65a2a0d98e69d0f7/35c3d94d0b4395d36b37c20b65a2a0d98e69d0f7

teto commented:
I made some progress on the gpgme python packaging but it still fails in the postinstallation phase with 
```
glibPreFixupPhase
post-installation fixup
Moving /nix/store/jzm6iwps6lys6vfxwpn7pnw976gy6011-gpgme-1.10.0/lib/cmake to /nix/store/vv1d1469zvh4mc0hgdp3r0fmba52qs12-gpgme-1.10.0-dev/lib/cmake
rmdir: failed to remove '/nix/store/jzm6iwps6lys6vfxwpn7pnw976gy6011-gpgme-1.10.0/lib': Directory not empty
shrinking RPATHs of ELF executables and libraries in /nix/store/jzm6iwps6lys6vfxwpn7pnw976gy6011-gpgme-1.10.0
shrinking /nix/store/jzm6iwps6lys6vfxwpn7pnw976gy6011-gpgme-1.10.0/lib/libgpgme.so.11.19.0
shrinking /nix/store/jzm6iwps6lys6vfxwpn7pnw976gy6011-gpgme-1.10.0/lib/libgpgmepp.so.6.5.0
strip is /nix/store/mdyy001q67hiks0g24ra53z7ckm4jfr4-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/jzm6iwps6lys6vfxwpn7pnw976gy6011-gpgme-1.10.0/lib 
patching script interpreter paths in /nix/store/jzm6iwps6lys6vfxwpn7pnw976gy6011-gpgme-1.10.0
checking for references to /tmp/nix-build-gpgme-1.10.0.drv-0 in /nix/store/jzm6iwps6lys6vfxwpn7pnw976gy6011-gpgme-1.10.0...
shrinking RPATHs of ELF executables and libraries in /nix/store/vv1d1469zvh4mc0hgdp3r0fmba52qs12-gpgme-1.10.0-dev
shrinking /nix/store/vv1d1469zvh4mc0hgdp3r0fmba52qs12-gpgme-1.10.0-dev/bin/gpgme-tool
strip is /nix/store/mdyy001q67hiks0g24ra53z7ckm4jfr4-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/vv1d1469zvh4mc0hgdp3r0fmba52qs12-gpgme-1.10.0-dev/lib  /nix/store/vv1d1469zvh4mc0hgdp3r0fmba52qs12-gpgme-1.10.0-dev/bin 
patching script interpreter paths in /nix/store/vv1d1469zvh4mc0hgdp3r0fmba52qs12-gpgme-1.10.0-dev
/nix/store/vv1d1469zvh4mc0hgdp3r0fmba52qs12-gpgme-1.10.0-dev/bin/gpgme-config: interpreter directive changed from "/bin/sh" to "/nix/store/i0ay05pqkbnvpfijm52mmlrp6kmkl80c-bash-4.4-p12/bin/sh"
checking for references to /tmp/nix-build-gpgme-1.10.0.drv-0 in /nix/store/vv1d1469zvh4mc0hgdp3r0fmba52qs12-gpgme-1.10.0-dev...
shrinking RPATHs of ELF executables and libraries in /nix/store/lhfzgh30949nhh9mdw3gnawagi4h7j5z-gpgme-1.10.0-info
strip is /nix/store/mdyy001q67hiks0g24ra53z7ckm4jfr4-binutils-2.28.1/bin/strip
patching script interpreter paths in /nix/store/lhfzgh30949nhh9mdw3gnawagi4h7j5z-gpgme-1.10.0-info
checking for references to /tmp/nix-build-gpgme-1.10.0.drv-0 in /nix/store/lhfzgh30949nhh9mdw3gnawagi4h7j5z-gpgme-1.10.0-info...
Obtaining file:///home/teto
    Complete output from command python setup.py egg_info:
    running egg_info
    creating gpg.egg-info
    writing gpg.egg-info/PKG-INFO
    writing dependency_links to gpg.egg-info/dependency_links.txt
    writing top-level names to gpg.egg-info/top_level.txt
    writing manifest file 'gpg.egg-info/SOURCES.txt'
    error: package directory 'gpg' does not exist
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /home/teto/
```
I dont know what invokes the egg_info, I grepped for it in nixpkgs to no avail. @FRidh any idea ? I use for my tests a local src if that helps ` src = /home/teto/gpgme;`.

fpletz commented on pkgs/development/libraries/gpgme/default.nix:
```diff
@@ -1,15 +1,21 @@
+  # src = fetchurl {
+  #   url = "mirror://gnupg/gpgme/${name}.tar.bz2";
+  #   sha256 = "14q619lxbk64vz7lih5gjb928qm28jrnn1h3yhsrrff3jw8yv3qs";
+  # };
```
What's going on here?

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,62 @@
+{ stdenv, lib
+, pythonPackages, fetchFromGitHub, notmuch, file, gpgme
+, withManpage ? true }:
```
This can be removed. It hardly takes any time to produce the manpages so we should.

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,62 @@
+  ];
+
+  # 3 tests fail, some unicode
+  doCheck = false;
```
So something may be broken?

FRidh commented on pkgs/development/libraries/gpgme/default.nix:
```diff
@@ -18,10 +24,15 @@ stdenv.mkDerivation rec {
+  nativeBuildInputs = [ pkgconfig gnupg autoreconfHook git texinfo5 ];
+
+  preConfigure = ''
+    ./autogen.sh
```
autoreconfHook?

teto commented:
I can't manage to get a wheel. Even if I could I am not sure how to write the package are the C parts 1 python part are intertwined. Any help would be greatly appreciated. My PR is a bit of a mess but basically

`nix-shell -A pythonPackages.gpg ~/nixpkgs2 --show-trace -K
unpackPhase
cd source
patchPhase
autoreconfPhase
configurePhase (with the flag     "--enable-languages=python")
cd lang/python
make all-local will build some python stuff
`
run the gpgme 

I've tried to add a target `all-bdist` to build the wheels at https://github.com/teto/gpgme/commits/nix but it still fails with
```
    installing to build/bdist.linux-x86_64/wheel
    running install
    Traceback (most recent call last):
      File "nix_run_setup", line 8, in <module>
        exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
      File "setup.py", line 275, in <module>
        'Topic :: Security :: Cryptography',
      File "/nix/store/xk2drqkvcv0cabgb3hsikh5vm5mhi9qi-python-2.7.14/lib/python2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/nix/store/xk2drqkvcv0cabgb3hsikh5vm5mhi9qi-python-2.7.14/lib/python2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/nix/store/xk2drqkvcv0cabgb3hsikh5vm5mhi9qi-python-2.7.14/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/nix/store/6hm110mhsj9nir4sg44j74a2fnnd77zf-python2.7-wheel-0.30.0/lib/python2.7/site-packages/wheel/bdist_wheel.py", line 257, in run
        self.distinfo_dir)
      File "/nix/store/6hm110mhsj9nir4sg44j74a2fnnd77zf-python2.7-wheel-0.30.0/lib/python2.7/site-packages/wheel/bdist_wheel.py", line 393, in egg2dist
        raise ValueError(err)
    ValueError: Egg metadata expected at build/bdist.linux-x86_64/wheel/gpg-1.10.1-py2.7.egg-info but not found
    make: *** [Makefile:761: all-bdist] Error 1

```

teto commented:
I succeeded in building a wheel but first I had to generate an egg (I thought wheels were supposed to replace eggs ? )  but the wheel looks empty, it just contains metadata. so the install process is smooth but once running `import gpg`, nothing happens since there are no gpg headers
```
/nix/store/l8fv56j5162iymyv9514vnnjr10kfvyg-python2.7-gpg-1.10.0/lib/python2.7/site-packages:
gpg-1.10.1.dist-info

/nix/store/l8fv56j5162iymyv9514vnnjr10kfvyg-python2.7-gpg-1.10.0/lib/python2.7/site-packages/gpg-1.10.1.dist-info:
DESCRIPTION.rst  INSTALLER  METADATA  metadata.json  RECORD  top_level.txt  WHEEL
```
I've tried other Makefile targets install-exec-local without success.

teto commented:
After an advice on gpg-users mailing list I gave up on building a separate package/wheel and just used the gpgme make install. I haven't tested the gpg part of alot (never used before) but reading mail works. It seems the manpage disappeared when upgrading to 0.7 but I don't have the time to look into it right now so I mark this rfc/rdy.

teto committed:
gpgme: add withPython ? false to enable bindings

Useful for MUAs mostly, like alot

teto committed:
service_identity: cosmetic changes

teto committed:
alot: 0.5.1 to 0.7.0

- added gnupg to checkInputs
- generate manpage (optional)
- move alot.desktop file to $out/share/applications
- disabled tests as they need the network (dependency on twisted)

Thanks to Sarah Brofeldt, Ben Mcginnes for their help (and other) and
to FRidh for the repeated careful reviews.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=9f40d187-dfcc-4b69-8378-ddd7f1e06552)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/lm2n7jppx7b7fn436b8548x366ixka8f-python2.7-alot-0.7
checking for references to /build in /nix/store/lm2n7jppx7b7fn436b8548x366ixka8f-python2.7-alot-0.7...
shrinking RPATHs of ELF executables and libraries in /nix/store/kpfaxc5xyc9qb0bnp86q86xbmb34l1rs-python2.7-alot-0.7-man
gzipping man pages under /nix/store/kpfaxc5xyc9qb0bnp86q86xbmb34l1rs-python2.7-alot-0.7-man/share/man/
strip is /nix/store/b0zlxla7dmy1iwc3g459rjznx59797xy-binutils-2.28.1/bin/strip
patching script interpreter paths in /nix/store/kpfaxc5xyc9qb0bnp86q86xbmb34l1rs-python2.7-alot-0.7-man
checking for references to /build in /nix/store/kpfaxc5xyc9qb0bnp86q86xbmb34l1rs-python2.7-alot-0.7-man...
wrapping `/nix/store/lm2n7jppx7b7fn436b8548x366ixka8f-python2.7-alot-0.7/bin/.alot-wrapped'...
/nix/store/lm2n7jppx7b7fn436b8548x366ixka8f-python2.7-alot-0.7
/nix/store/a7idzs9ryxhkrbwl1lnwvhj2xin8nziw-gpgme-1.10.0
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=067d15ef-055f-488b-a797-d226e5dd5476)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/7jsr93skjk0qpnnwdivinrsrql2gmls9-python2.7-alot-0.7
checking for references to /build in /nix/store/7jsr93skjk0qpnnwdivinrsrql2gmls9-python2.7-alot-0.7...
shrinking RPATHs of ELF executables and libraries in /nix/store/6zpriyhca33h23y17vszxjsb7llmi96a-python2.7-alot-0.7-man
gzipping man pages under /nix/store/6zpriyhca33h23y17vszxjsb7llmi96a-python2.7-alot-0.7-man/share/man/
strip is /nix/store/lvx1acn1ig1j2km8jds5x3ggh3f2wa8v-binutils-2.28.1/bin/strip
patching script interpreter paths in /nix/store/6zpriyhca33h23y17vszxjsb7llmi96a-python2.7-alot-0.7-man
checking for references to /build in /nix/store/6zpriyhca33h23y17vszxjsb7llmi96a-python2.7-alot-0.7-man...
wrapping `/nix/store/7jsr93skjk0qpnnwdivinrsrql2gmls9-python2.7-alot-0.7/bin/.alot-wrapped'...
/nix/store/7jsr93skjk0qpnnwdivinrsrql2gmls9-python2.7-alot-0.7
/nix/store/d909mg95hlgqb8hxqhsxmr2j0434x45r-gpgme-1.10.0
```
</p></details>

teto commented:
I just wanted to check  if I could reenable the tests  from a nix-shell with `export out=toto` but these 2 simple tests still fail https://github.com/pazz/alot/blob/master/tests/helper_test.py#L442
```
test_simple_address_with_real_name (tests.completion_test.AbooksCompleterTest) ... ok
test_dont_choke_on_special_regex_characters (tests.completion_test.StringlistCompleterTest) ... ok

======================================================================
ERROR: test_env_doesnt_pollute (tests.helper_test.TestCallCmdAsync)
test_env_doesnt_pollute
----------------------------------------------------------------------
ProcessTerminated: A process has ended with a probable error condition: process ended with exit code 1.

======================================================================
ERROR: test_env_set (tests.helper_test.TestCallCmdAsync)
test_env_set
----------------------------------------------------------------------
ProcessTerminated: A process has ended with a probable error condition: process ended with exit code 1.

----------------------------------------------------------------------
Ran 340 tests in 6.602s

FAILED (errors=2, expected failures=2)
Test failed: <unittest.runner.TextTestResult run=340 errors=2 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=340 errors=2 failures=0>

nix-shell -A python27Packages.alot ~/nixpkgs  10,03s user 1,82s system 43% cpu 27,462 total
```

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=a9622772-4ded-41ff-97a1-6a5b84d34bd4)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
AssertionError: Expected 'notify' to not have been called. Called 1 times.

----------------------------------------------------------------------
Ran 340 tests in 13.025s

FAILED (errors=3, expected failures=2)
Test failed: <unittest.runner.TextTestResult run=340 errors=3 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=340 errors=3 failures=0>
builder for '/nix/store/671hydwkwy93cgabbz2vx3f0wl43gkk9-python2.7-alot-0.7.drv' failed with exit code 1
[31;1merror:[0m build of '/nix/store/671hydwkwy93cgabbz2vx3f0wl43gkk9-python2.7-alot-0.7.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=9747a113-fb47-4acb-992b-90a912c542e6)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
OSError: [Errno 2] No such file or directory: '/tmp/nix-build-python2.7-alot-0.7.drv-0/tmp5UXmy4/S.gpg-agent.extra'

----------------------------------------------------------------------
Ran 340 tests in 5.004s

FAILED (errors=5, expected failures=2)
Test failed: <unittest.runner.TextTestResult run=340 errors=5 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=340 errors=5 failures=0>
builder for ‘/nix/store/5ll1ah7di63r2dk3fxfggazg0rqvmlhr-python2.7-alot-0.7.drv’ failed with exit code 1
error: build of ‘/nix/store/5ll1ah7di63r2dk3fxfggazg0rqvmlhr-python2.7-alot-0.7.drv’ failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=86b459e9-da89-405b-a38f-79c5548dcfdb)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/9hi9dm6lnjnxh9l94rxb44wr7f8dgr1f-python2.7-alot-0.7
checking for references to /build in /nix/store/9hi9dm6lnjnxh9l94rxb44wr7f8dgr1f-python2.7-alot-0.7...
shrinking RPATHs of ELF executables and libraries in /nix/store/jj9z3325d1q9997h0mjdgw52b00zxn0i-python2.7-alot-0.7-man
gzipping man pages under /nix/store/jj9z3325d1q9997h0mjdgw52b00zxn0i-python2.7-alot-0.7-man/share/man/
strip is /nix/store/b0zlxla7dmy1iwc3g459rjznx59797xy-binutils-2.28.1/bin/strip
patching script interpreter paths in /nix/store/jj9z3325d1q9997h0mjdgw52b00zxn0i-python2.7-alot-0.7-man
checking for references to /build in /nix/store/jj9z3325d1q9997h0mjdgw52b00zxn0i-python2.7-alot-0.7-man...
wrapping `/nix/store/9hi9dm6lnjnxh9l94rxb44wr7f8dgr1f-python2.7-alot-0.7/bin/.alot-wrapped'...
/nix/store/9hi9dm6lnjnxh9l94rxb44wr7f8dgr1f-python2.7-alot-0.7
/nix/store/a7idzs9ryxhkrbwl1lnwvhj2xin8nziw-gpgme-1.10.0
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=bcf609a4-04c7-4bb0-b7ab-1947ed3b0296)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/034rff7qm6vs4ybs70nmvahw5mg3sh4k-python2.7-alot-0.7
checking for references to /build in /nix/store/034rff7qm6vs4ybs70nmvahw5mg3sh4k-python2.7-alot-0.7...
shrinking RPATHs of ELF executables and libraries in /nix/store/b91hf6ypz2xkbwzl3413zzldv437vbh9-python2.7-alot-0.7-man
gzipping man pages under /nix/store/b91hf6ypz2xkbwzl3413zzldv437vbh9-python2.7-alot-0.7-man/share/man/
strip is /nix/store/lvx1acn1ig1j2km8jds5x3ggh3f2wa8v-binutils-2.28.1/bin/strip
patching script interpreter paths in /nix/store/b91hf6ypz2xkbwzl3413zzldv437vbh9-python2.7-alot-0.7-man
checking for references to /build in /nix/store/b91hf6ypz2xkbwzl3413zzldv437vbh9-python2.7-alot-0.7-man...
wrapping `/nix/store/034rff7qm6vs4ybs70nmvahw5mg3sh4k-python2.7-alot-0.7/bin/.alot-wrapped'...
/nix/store/034rff7qm6vs4ybs70nmvahw5mg3sh4k-python2.7-alot-0.7
/nix/store/d909mg95hlgqb8hxqhsxmr2j0434x45r-gpgme-1.10.0
```
</p></details>

teto commented:
I think it's LGTM to disable the tests, it seems like they rely on twisted functions which need the network, hence the failures (twisted tests were disabled for that reason).

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=8dba6a58-53ba-435b-b852-2d7302bcaf0b)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/nn1j6paanw073r8q411gcx8zwn5wnvhy-python2.7-alot-0.7
checking for references to /tmp/nix-build-python2.7-alot-0.7.drv-0 in /nix/store/nn1j6paanw073r8q411gcx8zwn5wnvhy-python2.7-alot-0.7...
shrinking RPATHs of ELF executables and libraries in /nix/store/fjaqhmgc7yxqbyqjlaf2swqwla6xpxlr-python2.7-alot-0.7-man
gzipping man pages under /nix/store/fjaqhmgc7yxqbyqjlaf2swqwla6xpxlr-python2.7-alot-0.7-man/share/man/
strip is /nix/store/b0zlxla7dmy1iwc3g459rjznx59797xy-binutils-2.28.1/bin/strip
patching script interpreter paths in /nix/store/fjaqhmgc7yxqbyqjlaf2swqwla6xpxlr-python2.7-alot-0.7-man
checking for references to /tmp/nix-build-python2.7-alot-0.7.drv-0 in /nix/store/fjaqhmgc7yxqbyqjlaf2swqwla6xpxlr-python2.7-alot-0.7-man...
wrapping `/nix/store/nn1j6paanw073r8q411gcx8zwn5wnvhy-python2.7-alot-0.7/bin/.alot-wrapped'...
/nix/store/nn1j6paanw073r8q411gcx8zwn5wnvhy-python2.7-alot-0.7
/nix/store/5dq6wrkh2vh4kf8krv4hrd4bb27agx9i-gpgme-1.10.0
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=01a5c398-d30e-4071-a154-323d5bc9e224)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/1kldswfcqxsv5dz2x0lfrhnhnfr76mbw-python2.7-alot-0.7
checking for references to /build in /nix/store/1kldswfcqxsv5dz2x0lfrhnhnfr76mbw-python2.7-alot-0.7...
shrinking RPATHs of ELF executables and libraries in /nix/store/9zrzpmwr5p3b306djdlhxzj3m3ch3qzj-python2.7-alot-0.7-man
gzipping man pages under /nix/store/9zrzpmwr5p3b306djdlhxzj3m3ch3qzj-python2.7-alot-0.7-man/share/man/
strip is /nix/store/lvx1acn1ig1j2km8jds5x3ggh3f2wa8v-binutils-2.28.1/bin/strip
patching script interpreter paths in /nix/store/9zrzpmwr5p3b306djdlhxzj3m3ch3qzj-python2.7-alot-0.7-man
checking for references to /build in /nix/store/9zrzpmwr5p3b306djdlhxzj3m3ch3qzj-python2.7-alot-0.7-man...
wrapping `/nix/store/1kldswfcqxsv5dz2x0lfrhnhnfr76mbw-python2.7-alot-0.7/bin/.alot-wrapped'...
/nix/store/1kldswfcqxsv5dz2x0lfrhnhnfr76mbw-python2.7-alot-0.7
/nix/store/vwn1idrl8cvfcvkh2wcnxb8z2id6k3hb-gpgme-1.10.0
```
</p></details>

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,64 @@
+buildPythonApplication rec {
+  pname = "alot";
+  version = "0.7";
+  outputs = [ "out" "man" ];
```
`man` optional?

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,64 @@
+
+  postInstall = ''
+    mkdir -p $out/share/applications $out/man
+    cp -r docs/build/man $out/man
```
`man` optional?

FRidh commented on pkgs/development/libraries/gpgme/default.nix:
```diff
@@ -18,11 +26,21 @@ stdenv.mkDerivation rec {
+  ++ lib.optionals withPython [ python swig2 which ];
+
+  preConfigure = ''
+    ./autogen.sh
```
is this needed with the `autoreconfHook`?

teto replied:
not anymore indeed.

FRidh commented on pkgs/applications/networking/mailreaders/alot/default.nix:
```diff
@@ -0,0 +1,64 @@
+
+    sed "s,/usr/bin,$out/bin,g" extra/alot.desktop > $out/share/applications/alot.desktop
+    wrapProgram $out/bin/alot \
+      --prefix LD_LIBRARY_PATH : '${lib.makeLibraryPath [ notmuch file gpgme ]}'
```
these should be patched in the code

teto replied:
leftover from the actual alot code. it seems to work fine without it so I removed it

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=b543d359-6570-4be7-9c2e-d4f0ccb883b8)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
glibPreFixupPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/f42bfw7y80gj57lc82bmqid1i2xh6g6j-python2.7-alot-0.7
strip is /nix/store/fzcs0fn6bb04m82frhlb78nc03ny3w55-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/f42bfw7y80gj57lc82bmqid1i2xh6g6j-python2.7-alot-0.7/lib  /nix/store/f42bfw7y80gj57lc82bmqid1i2xh6g6j-python2.7-alot-0.7/bin
patching script interpreter paths in /nix/store/f42bfw7y80gj57lc82bmqid1i2xh6g6j-python2.7-alot-0.7
checking for references to /tmp/nix-build-python2.7-alot-0.7.drv-0 in /nix/store/f42bfw7y80gj57lc82bmqid1i2xh6g6j-python2.7-alot-0.7...
wrapping `/nix/store/f42bfw7y80gj57lc82bmqid1i2xh6g6j-python2.7-alot-0.7/bin/alot'...
/nix/store/f42bfw7y80gj57lc82bmqid1i2xh6g6j-python2.7-alot-0.7
/nix/store/j3bqsky2k73ixbpf1dgxkrjk75fl7cna-gpgme-1.10.0
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=9c7b4848-1f0f-43cb-8c5d-8b74cf86917a)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
   Compiling rustc_passes v0.0.0 (file:///build/rustc-1.24.0-src/src/librustc_passes)
building of '/nix/store/6mm6pr840z429p4l9mk62gh7mr6khj3k-rustc-1.24.0.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/wsycxxpy4fin6rmfmx67n1b5pdzjww20-cargo-0.24.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/byqkqqkjq1m38ikbxxrirk81d0wr4qns-librsvg-2.42.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/azb669rka042skzps2bpc7m7bs1r5afm-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/z39d6dyfqhk47n5ak0l39jfr4byd973r-emacs-25.3.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/647mvp5wbl77a34wgysybs2m0j3d5ijv-notmuch-0.26.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/10jnxjdm5z0ixsi831cswkp28mbkh5z2-python2.7-python-notmuch-0.26.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/kbfk3hz0cp9kf33zq88rbr641j7m5m9j-python2.7-alot-0.7.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/kbfk3hz0cp9kf33zq88rbr641j7m5m9j-python2.7-alot-0.7.drv' failed
```
</p></details>

Mic92 commented:
@GrahamcOfBorg build alot gpgme

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=2500d8ad-c79e-4e89-97c9-cdc7084147f7)

Attempted: gpgme

The following builds were skipped because they don't evaluate on x86_64-darwin: alot

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/vb6mgjqgd9h11nbv60fvrp1ls9nxck2y-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/akiwdsbny6qk0dww3inwxx7dz9853i8n-gpgme-1.10.0/lib
patching script interpreter paths in /nix/store/akiwdsbny6qk0dww3inwxx7dz9853i8n-gpgme-1.10.0
strip is /nix/store/vb6mgjqgd9h11nbv60fvrp1ls9nxck2y-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/wiyfc23yaz7fhnzxyqiswwjyq8kglyz9-gpgme-1.10.0-dev/lib  /nix/store/wiyfc23yaz7fhnzxyqiswwjyq8kglyz9-gpgme-1.10.0-dev/bin
patching script interpreter paths in /nix/store/wiyfc23yaz7fhnzxyqiswwjyq8kglyz9-gpgme-1.10.0-dev
/nix/store/wiyfc23yaz7fhnzxyqiswwjyq8kglyz9-gpgme-1.10.0-dev/bin/gpgme-config: interpreter directive changed from "/bin/sh" to "/nix/store/8hdxjv3m14jgdnawi5l0mz2h59qkk2g9-bash-4.4-p19/bin/sh"
strip is /nix/store/vb6mgjqgd9h11nbv60fvrp1ls9nxck2y-cctools-binutils-darwin/bin/strip
patching script interpreter paths in /nix/store/0fxwndpkffnssk23syskklk9gvpkpjmi-gpgme-1.10.0-info
/nix/store/akiwdsbny6qk0dww3inwxx7dz9853i8n-gpgme-1.10.0
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=24ab7340-37d7-49f1-a124-9861d3f7bd64)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
glibPreFixupPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7
strip is /nix/store/fzcs0fn6bb04m82frhlb78nc03ny3w55-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7/lib  /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7/bin
patching script interpreter paths in /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7
checking for references to /tmp/nix-build-python2.7-alot-0.7.drv-0 in /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7...
wrapping `/nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7/bin/alot'...
/nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7
/nix/store/917ir2mcsdvi45gnlyv153ngvmnl59lq-gpgme-1.10.0
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=385f5317-f230-4de1-a123-f01baeade6c1)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
glibPreFixupPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/l6j4m4518fd969w7l68psw5a8v9d8ps1-python2.7-alot-0.7
strip is /nix/store/3zq400fri5dv7d30lpxlqm2v9y1iis6j-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/l6j4m4518fd969w7l68psw5a8v9d8ps1-python2.7-alot-0.7/lib  /nix/store/l6j4m4518fd969w7l68psw5a8v9d8ps1-python2.7-alot-0.7/bin
patching script interpreter paths in /nix/store/l6j4m4518fd969w7l68psw5a8v9d8ps1-python2.7-alot-0.7
checking for references to /build in /nix/store/l6j4m4518fd969w7l68psw5a8v9d8ps1-python2.7-alot-0.7...
wrapping `/nix/store/l6j4m4518fd969w7l68psw5a8v9d8ps1-python2.7-alot-0.7/bin/alot'...
/nix/store/l6j4m4518fd969w7l68psw5a8v9d8ps1-python2.7-alot-0.7
/nix/store/0clchyfabn9cvv2qh1m0anhq5hmn25hb-gpgme-1.10.0
```
</p></details>

teto commented:
Seems to pass :D

Mic92 committed:
python.pkgs.gpg: rename to gpgme

Let's be honest with what we expose.

Mic92 commented on pkgs/top-level/python-packages.nix:
```diff
@@ -8173,7 +8173,7 @@ in {
   google_gax = callPackage ../development/python-modules/google_gax { };
 
-  gpg = toPythonModule (pkgs.gpgme.override { withPython=true; });
+  gpgme = toPythonModule (pkgs.gpgme.override { withPython=true; });
```
@teto do you agree with change?

teto replied:
I hesitated between the 2 and ended up with `gpg` because it's `import gpg ` and it's like the official ports for gpg (even if gpgme expose a simpler interface). So I have no problem with changing the name :)

Mic92 commented:
@GrahamcOfBorg build alot gpgme

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=d8d61041-5569-4bf5-ae45-c5d67cf254d6)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
glibPreFixupPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7
strip is /nix/store/fzcs0fn6bb04m82frhlb78nc03ny3w55-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7/lib  /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7/bin
patching script interpreter paths in /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7
/nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7
checking for references to /build in /nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7...
/nix/store/917ir2mcsdvi45gnlyv153ngvmnl59lq-gpgme-1.10.0
wrapping `/nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7/bin/alot'...
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=ddf29311-2411-441f-8410-b7654cc0f1a4)

Attempted: gpgme

The following builds were skipped because they don't evaluate on x86_64-darwin: alot

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/akiwdsbny6qk0dww3inwxx7dz9853i8n-gpgme-1.10.0
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=76e902c7-3af1-4c11-b17b-add6723a9402)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/7v8nzwm05jb708ws1yh7dy0icv94n81h-python2.7-alot-0.7
/nix/store/917ir2mcsdvi45gnlyv153ngvmnl59lq-gpgme-1.10.0
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=680d2e40-c72d-4c9d-a43e-ab037e796211)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
   Compiling rustc_mir v0.0.0 (file:///build/rustc-1.24.0-src/src/librustc_mir)
building of '/nix/store/6mm6pr840z429p4l9mk62gh7mr6khj3k-rustc-1.24.0.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/68kg61kq3qlbfxdjan2s7hrd0fidyxri-cargo-0.24.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/2fvhpqf8jjcdzvifkpipb5r41pwvxl2d-librsvg-2.42.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/92wjazfxwjpwv6zkqr6xkdxcy1illhny-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/crcmgcgk9q8s2mjj595hn5104xiwdhvi-emacs-25.3.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/n1860wkakmh2hzx3aaqpivbkaf9ifm4l-notmuch-0.26.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/iw0hhzajgwd0x2cgpsbacrwsgn98x7sv-python2.7-python-notmuch-0.26.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/a6lzm8azz82y8lrdqqr18j77jxjpch6n-python2.7-alot-0.7.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/a6lzm8azz82y8lrdqqr18j77jxjpch6n-python2.7-alot-0.7.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.30429&attempt_id=0cab4aa6-fcba-4475-a50b-0c654da1a8cf)

Attempted: alot, gpgme

<details><summary>Partial log (click to expand)</summary><p>

```
   Compiling backtrace v0.3.4
building of '/nix/store/6mm6pr840z429p4l9mk62gh7mr6khj3k-rustc-1.24.0.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/68kg61kq3qlbfxdjan2s7hrd0fidyxri-cargo-0.24.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/2fvhpqf8jjcdzvifkpipb5r41pwvxl2d-librsvg-2.42.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/92wjazfxwjpwv6zkqr6xkdxcy1illhny-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/crcmgcgk9q8s2mjj595hn5104xiwdhvi-emacs-25.3.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/n1860wkakmh2hzx3aaqpivbkaf9ifm4l-notmuch-0.26.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/iw0hhzajgwd0x2cgpsbacrwsgn98x7sv-python2.7-python-notmuch-0.26.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/a6lzm8azz82y8lrdqqr18j77jxjpch6n-python2.7-alot-0.7.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/a6lzm8azz82y8lrdqqr18j77jxjpch6n-python2.7-alot-0.7.drv' failed
```
</p></details>

teto commented:
thanks for the keeping up with the good reviews despite my numerous hesitations in the implementation. Really happy to see it merged !

Mic92 commented:
The show must go on.
